tests/unit/coordinator/test_list.py

```diff
@@ -71,3 +93,207 @@ def _run():
+                                "store/000/00000000-0000-0000-0000-100000000002/detached/all_0_0_0/data.bin"),
+                            file_size=1000,
+                            mtime_ns=0,
+                            hexdigest="2000",
+                        ),
+                        SnapshotFile(
+                            relative_path=Path(
+                                "store/000/00000000-0000-0000-0000-100000000002/detached/all_1_1_0/data.bin"),
+                            file_size=1000,
+                            mtime_ns=0,
+                            hexdigest="2110",
+                        ),
+                    ],
+                ),
+                hashes=[
+                    SnapshotHash(hexdigest="1000", size=1000),
+                    SnapshotHash(hexdigest="1110", size=1000),
+                    SnapshotHash(hexdigest="2000", size=1000),
+                    SnapshotHash(hexdigest="2110", size=1000),
+                ],
+                files=4,
+                total_size=4000,
+            ),
+        ],
+        upload_results=[
+            SnapshotUploadResult(total_size=4000, total_stored_size=3000),
+            SnapshotUploadResult(total_size=4000, total_stored_size=3000),
+        ],
+        plugin=Plugin.clickhouse,
+    )
+
+
+@pytest.fixture(name="backup_manifest_nostate")
+def fixture_backup_manifest_nostate() -> BackupManifest:
+    """Provide a backup manifest that's missing the snapshot state."""
+    return BackupManifest(
+        start=datetime.datetime(2020, 1, 2, 3, 4, 5, 678, tzinfo=datetime.timezone.utc),
+        end=datetime.datetime(2020, 1, 2, 5, 6, 7, 891, tzinfo=datetime.timezone.utc),
+        attempt=1,
+        snapshot_results=[
+            # First node
+            SnapshotResult(
+                state=None,
+                hashes=[
+                    SnapshotHash(hexdigest="1000", size=1000),
+                    SnapshotHash(hexdigest="1110", size=1000),
+                    SnapshotHash(hexdigest="1100", size=1000),
+                    SnapshotHash(hexdigest="2000", size=1000),
+                ],
+                files=4,
+                total_size=4000,
+            ),
+            # Second node
+            SnapshotResult(
+                state=None,
+                hashes=[
+                    SnapshotHash(hexdigest="1000", size=1000),
+                    SnapshotHash(hexdigest="1110", size=1000),
+                    SnapshotHash(hexdigest="2000", size=1000),
+                    SnapshotHash(hexdigest="2110", size=1000),
+                ],
+                files=4,
+                total_size=4000,
+            ),
+        ],
+        upload_results=[
+            SnapshotUploadResult(total_size=4000, total_stored_size=3000),
+            SnapshotUploadResult(total_size=4000, total_stored_size=3000),
+        ],
+        plugin=Plugin.cassandra,
+    )
+
+
+def test_compute_unique_snapshot_file_stats(backup_manifest: BackupManifest):
```

Comment:
nit. return type is missing (next function too).

Reply:
I wish I could force mypy to do it, it's really not intuitive that "nothing" means "anything" instead of None.

(Which, in case of tests, becomes fun when you accidentally `return expected` instead of `assert expected` and mypy doesn't complain).

Reply:
There is an option to force typing, it is not just set for astacus as I didn't have time for it, see mypy.ini:

```
#disallow_incomplete_defs = true
#disallow_untyped_defs = true
```

Reply:
IIUC enabling these will most probably main a deluge of errors in the future, whenever an untyped module is touched.

Reply:
Well, those should really be enabled in some future PR which does it for whole codebase. (I had partial effort but it turned out to be bit much).  Probably starting with incomplete defs would help, and then later on doing all untyped ones.